servo/repeating.py

```diff
@@ -62,13 +71,28 @@ def start_repeating_task(
 
         async def repeating_async_fn() -> None:
             while True:
-                callable()
-                await asyncio.sleep(every.total_seconds())
+                t0 = time.time()
+                if asyncio.iscoroutinefunction(function):
+                    await function()
+                elif callable(function):
+                    function()
+                else:
+                    raise TypeError(f"function={function} must be Awaitable or Callable, but has "
+                        f"type(function)={type(function)}.")
+                t1 = time.time()
+                sleep_time = every.total_seconds()
+                if time_correction:
+                    sleep_time -= max(t1 - t0, 0)
```

Comment:
This should be `sleep_time = max(sleep_time - (t1 - t0), 0)` or drop the `max` since you check for < 0 below and time can only flow forward :wink:

Reply:
Nope - this reads more clearly as is, the functionality is identical, and the purpose of a review is not to nitpick specific implementation details (especially if they are redundant) but to make sure there aren't errors the tests don't catch.